pkg/operator/sync.go

```diff
@@ -206,15 +227,23 @@ func (optr *Operator) syncRenderConfig(_ *renderConfig) error {
 	}
 	spec.AdditionalTrustBundle = trustBundle
 
-	// if the cloudConfig is set in infra read the cloud config reference
-	if infra.Spec.CloudConfig.Name != "" {
-		cc, err := optr.getCloudConfigFromConfigMap("openshift-config", infra.Spec.CloudConfig.Name, infra.Spec.CloudConfig.Key)
+	// check if openshift-config-managed/kube-cloud-config ConfigMap is available and cloud config is required for given platform
+	if _, err = optr.clusterCmLister.ConfigMaps("openshift-config-managed").Get("kube-cloud-config"); err != nil {
+		// processCloudConfigMapError is mainly to keep cyclomatic complexity of syncRenderConfig function in range of 30
+		if ccErr := processCloudConfigMapError(err, infra); ccErr != nil {
+			return ccErr
+		}
```

Comment:
processCloudConfigMapError returns nil error if the err are was not `NotFound`, and therefore we are skipping handling of error like NotAuthorized for example.

Reply:
fixed! I wasn't sure what others error could be and if it makes sense to return errors in those cases or continue further.